### PR TITLE
Add step and offset parameters to the Number validator to allow discrete numeric ranges

### PR DESCRIPTION
**Pull request by confluence**
This should make it possible to use the `Number` validator to validate discrete number ranges with a fixed step size and offset. It should not affect existing usage of the validator. Validation doesn't check *type* -- if `1` is a valid value, so is `1.0`.

### CODE REVIEW

**Comment by confluence** [comment]
I have tested this in `set_channel` (which should have integer values, which we can now enforce), and it seems to be working.

(I have merged in the `set_channel` fix from #101 -- I'll merge dev in here after that's merged so that we have a clean diff.)

**Review by kswang1029 — APPROVED** [review]

**Review by I-Chenn — APPROVED** [review]

**Pull request merged**
